web-ng/src/app/components/map/map.component.ts

```diff
@@ -101,9 +129,13 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     this.subscription.unsubscribe();
   }
 
-  onProjectAndFeaturesUpdate(project: Project, features: List<Feature>) {
-    this.clearMap();
-    this.addMarkersAndGeoJsonsToMap(project, features);
+  onProjectAndFeaturesUpdate(
+    project: Project,
+    featuresToAdd: List<Feature>,
+    feautresToRemove: List<Feature>
+  ) {
+    this.removeMarkersAndGeoJsonsOnMap(feautresToRemove);
```

Comment:
Considering the above, instead of receiving the features to add and remove, we'd receive the latest List of features, and do a pairwise comparison, adding, removing, and updating as we go.

Reply:
Done.